stdlib/random.ml

```diff
@@ -40,6 +40,35 @@ module State = struct
     let s = create () in
     set s i1 i2 i3 i4; s
 
+  let serialization_prefix =
+    (* '1' is a version number; more than 200 versions may fit a char. *)
+    "lxm1:"
+  let serialization_prefix_len =
+    String.length serialization_prefix
+
+  let to_bytes s =
+    let prefix = serialization_prefix in
+    let preflen = serialization_prefix_len in
+    let buf = Bytes.create (preflen + 4 * 8) in
+    Bytes.blit_string prefix 0 buf 0 preflen;
+    for i = 0 to 3 do
+      Bytes.set_int64_le buf (preflen + i * 8) (Array1.get s i)
+    done;
+    buf
+
+  let of_bytes buf =
+    let prefix = serialization_prefix in
+    let preflen = serialization_prefix_len in
+    if Bytes.length buf <> preflen + (4 * 8)
+       || Bytes.sub_string buf 0 preflen <> prefix
```

Comment:
`Bytes.start_with`

Reply:
That's a good idea, I tried, it does not work:
1. the size check is more precise than what `starts_with` does (I check for both the prefix and the payload size), so I would need a size check anyway
2. `Bytes.starts_with` takes a `Bytes.t` as prefix parameter, I have a string, and adding a conversion obscures things. (I guess we should have `{starts,ends}_with_string`.

In the end it was simpler and clearer to leave the code as-is. Thanks anyways :-)